tests/test_unit_test_hocons.py

```diff
@@ -0,0 +1,61 @@
+
+# Copyright © 2023-2025 Cognizant Technology Solutions Corp, www.cognizant.com.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# END COPYRIGHT
+
+from unittest import TestCase
+
+import pytest
+
+from parameterized import parameterized
+
+from neuro_san.test.unittest.dynamic_hocon_unit_tests import DynamicHoconUnitTests
+
+
+class TestUnitTestHocons(TestCase):
+    """
+    Data-driven dynamic test cases where each test case is specified by a single hocon file.
+    """
+
+    # A single instance of the DynamicHoconUnitTests helper class.
+    # We pass it our source file location and a relative path to the common
+    # root of the test hocon files listed in the @parameterized.expand()
+    # annotation below so the instance can find the hocon test cases listed.
+    DYNAMIC = DynamicHoconUnitTests(__file__, path_to_basis="./fixtures")
+
+    @parameterized.expand(DynamicHoconUnitTests.from_hocon_list([
+        # These can be in any order.
+        # Ideally more basic functionality will come first.
+        # Barring that, try to stick to alphabetical order.
+
+        # Comment this back in when it's working more reliably.
+        # "multiagent_decomposer/readme_test_case.hocon",
```

Comment:
Test harness for running hoc on-based test cases for agents.
Currently the readme_test_case.hocon is not passing often enough to turn this test on.
A subsequent PR will fix this up and then this line gets uncommented.